ogr/ogrsf_frmts/gpkg/gdalgeopackagerasterband.cpp

```diff
@@ -1229,6 +1231,144 @@ CPLErr GDALGPKGMBTilesLikeRasterBand::IReadBlock(int nBlockXOff, int nBlockYOff,
     return CE_None;
 }
 
+/************************************************************************/
+/*                       IGetDataCoverageStatus()                       */
+/************************************************************************/
+
+int GDALGPKGMBTilesLikeRasterBand::IGetDataCoverageStatus(int nXOff, int nYOff,
+                                                          int nXSize,
+                                                          int nYSize,
+                                                          int nMaskFlagStop,
+                                                          double *pdfDataPct)
+{
+    if (eAccess == GA_Update)
+        FlushCache(false);
+
+    const int iColMin = nXOff / nBlockXSize + m_poTPD->m_nShiftXTiles;
+    const int iColMax = (nXOff + nXSize - 1) / nBlockXSize +
+                        m_poTPD->m_nShiftXTiles +
+                        (m_poTPD->m_nShiftXPixelsMod ? 1 : 0);
+    const int iRowMin = nYOff / nBlockYSize + m_poTPD->m_nShiftYTiles;
+    const int iRowMax = (nYOff + nYSize - 1) / nBlockYSize +
+                        m_poTPD->m_nShiftYTiles +
+                        (m_poTPD->m_nShiftYPixelsMod ? 1 : 0);
+
+    int iDBRowMin = m_poTPD->GetRowFromIntoTopConvention(iRowMin);
+    int iDBRowMax = m_poTPD->GetRowFromIntoTopConvention(iRowMax);
+    if (iDBRowMin > iDBRowMax)
+        std::swap(iDBRowMin, iDBRowMax);
+
+    char *pszSQL = sqlite3_mprintf(
+        "SELECT tile_row, tile_column FROM \"%w\" "
+        "WHERE zoom_level = %d AND "
+        "(tile_row BETWEEN %d AND %d) AND "
+        "(tile_column BETWEEN %d AND %d)"
+        "%s",
+        m_poTPD->m_osRasterTable.c_str(), m_poTPD->m_nZoomLevel, iDBRowMin,
+        iDBRowMax, iColMin, iColMax,
+        !m_poTPD->m_osWHERE.empty()
+            ? CPLSPrintf(" AND (%s)", m_poTPD->m_osWHERE.c_str())
+            : "");
+
+#ifdef DEBUG_VERBOSE
+    CPLDebug("GPKG", "%s", pszSQL);
+#endif
+
+    sqlite3_stmt *hStmt = nullptr;
+    int rc = sqlite3_prepare_v2(m_poTPD->IGetDB(), pszSQL, -1, &hStmt, nullptr);
+    if (rc != SQLITE_OK)
+    {
+        CPLError(CE_Failure, CPLE_AppDefined, "failed to prepare SQL %s: %s",
+                 pszSQL, sqlite3_errmsg(m_poTPD->IGetDB()));
+        sqlite3_free(pszSQL);
+        return GDAL_DATA_COVERAGE_STATUS_UNIMPLEMENTED |
```

Comment:
Is this the correct return value in case of errors? The doc say: `Flag returned by GDALGetDataCoverageStatus() when the driver does not implement GetDataCoverageStatus().` but here it is implemented.

Reply:
> Is this the correct return value in case of errors? The doc say: `Flag returned by GDALGetDataCoverageStatus() when the driver does not implement GetDataCoverageStatus().` but here it is implemented.

yeah, we don't really have a status code for error. So I thought defaulting to that was the best. With GDAL_DATA_COVERAGE_STATUS_UNIMPLEMENTED we document that the result isn't totally reliable. And with GDAL_DATA_COVERAGE_STATUS_DATA we are "pessimistic", so that user code doesn't skip that region